core/src/SYCL/Kokkos_SYCL_ParallelFor_Range.hpp

```diff
@@ -74,16 +74,15 @@ class Kokkos::Impl::ParallelFor<FunctorType, Kokkos::RangePolicy<Traits...>,
   const Policy m_policy;
 
   template <typename Functor>
-  static sycl::event sycl_direct_launch(const Policy& policy,
-                                        const Functor& functor,
-                                        const sycl::event& memcpy_event) {
+  sycl::event sycl_direct_launch(const Policy& policy, const Functor& functor,
```

Comment:
Just observing that this PR makes all the `Parallel<Construct>::sycl_direct_launch` have a somewhat misleading name because they now either launch a kernel or append a node to some graph.
Any thought on whether/how we can improve?

Reply:
Yeah, we could inline `sycl_direct_launch` into `execute` but that's just as misleading (but there is no good reason anymore not to do it other than that there is little gain from it).
Just as for all other Graph specializations, the implementation calls `execute` since that's where the functor is set up. In the case of `HIP` and `Cuda`, the functor is then passed to a launch mechanism outside, and only there treating it as a regular kernel or graph node happens. This won't work once a parallel construct requires more than one node though. 

We could think about renaming `execute` everywhere and split it into a `prepare_kernel` and a `launch` step and then introduce a `setup_graph_node` function (which might just call `launch` for Cuda and HIP but would call `sycl_attach_kernel_to_node` for SYCL).